qa/tasks/ceph_iscsi.py

```diff
@@ -0,0 +1,159 @@
+            remote.run(args=args, stdout=stdout, stderr=stderr)
+        except CommandFailedError:
+            if "No such file or directory" in stderr.getvalue():
+                raise
+
+        ips = re.findall(r'mon host = \[v2:([\d.]+).*v1.*\],\[v2:([\d.]+).*v1.*\],\[v2:([\d.]+).*v1.*\]', stdout.getvalue())
+        self.trusted_ip_list = list(ips[0])
+        if self.trusted_ip_list is []:
+            raise RuntimeError("no trust ip list was found!")
+
+    def run_daemon(self, remote, cmds):
+        p = remote.run(args=['sudo', 'adjust-ulimits', 'daemon-helper', 'kill', cmds],
+                          wait=False, stdin=run.PIPE, stdout=StringIO())
+        self.background_procs.append(p)
+
+    def _kill_background(self, p):
+        if p.stdin:
+            p.stdin.close()
+            try:
+                p.wait()
+            except (CommandFailedError, ConnectionLostError):
+                pass
+
+    def kill_backgrounds(self):
+        for p in self.background_procs:
+            self._kill_background(p)
+        self.background_procs = []
+
+    def _setup_gateway(self, role):
+        """Spawned task that setups the gateway"""
+        (remote,) = self.ctx.cluster.only(role).remotes.keys()
+
+        # setup the iscsi-gateway.cfg file, we only set the
+        # clust_name and trusted_ip_list and all the others
+        # as default
+        target = "/etc/ceph/ceph.conf"
+        self._get_trusted_ip_list(remote, target)
+
+        args = ['sudo', 'echo', 'cluster_name = ceph', run.Raw('>'), target]
+        remote.run(args=args)
+        args = ['sudo', 'echo', 'pool = rbd', run.Raw('>>'), target]
+        remote.run(args=args)
+        args = ['sudo', 'echo', 'api_secure = false', run.Raw('>>'), target]
+        remote.run(args=args)
+        args = ['sudo', 'echo', 'api_port = 5000', run.Raw('>>'), target]
+        remote.run(args=args)
+        ips = ','.join(self.trusted_ip_list)
+        args = ['sudo', 'echo', 'trusted_ip_list = {}'.format(ips), run.Raw('>>'), target]
+        remote.run(args=args)
+
+        self.run_daemon(remote, "/usr/bin/tcmu-runner")
+        self.run_daemon(remote, "/usr/bin/rbd-target-gw")
+        self.run_daemon(remote, "/usr/bin/rbd-target-api")
+
+    def setup_gateways(self):
+        for role in self.config['gateways']:
+            self._setup_gateway(role)
+
+    def _setup_client(self, role):
+        """Spawned task that setups the gateway"""
+        (remote,) = self.ctx.cluster.only(role).remotes.keys()
+
+        target = "/etc/iscsi/initiatorname.iscsi"
+        remote.run(args=['sudo', 'echo', f'InitiatorName={self.client_iqn}', run.Raw('>'), target])
+        # the restart is needed after the above change is applied
+        remote.run(args=['sudo', 'systemctl', 'restart', 'iscsid'])
+
+        remote.run(args=['sudo', 'modprobe', 'dm_multipath'])
+        remote.run(args=['sudo', 'mpathconf', '--enable'])
+        conf = dedent('''\
+devices {
```

Comment:
Use the `multipath` config from the documentation [1]

[1] https://docs.ceph.com/en/latest/rbd/iscsi-initiator-linux/

Reply:
Sure.